src/parser/monk/mistweaver/modules/talents/Upwelling.tsx

```diff
@@ -29,30 +31,32 @@ class Upwelling extends Analyzer {
     combatants: Combatants,
   };
 
-  totalHealing = 0;
-  totalOverhealing = 0;
-  totalAbsorbs = 0;
+  protected combatants!: Combatants;
 
-  castEF = 0;//casts of ef
-  extraHots = 0;//number of extra hots
-  extraBolts = 0;//number of extra bolts
-  efHotHeal = 0;//healing from hots
-  efHotOverheal = 0;//overhealing from hots
+  totalHealing: number = 0;
+  totalOverhealing: number = 0;
+  totalAbsorbs: number = 0;
 
-  boltCount = 0;
+  castEF: number = 0;//casts of ef
+  extraHots: number = 0;//number of extra hots
+  extraBolts: number = 0;//number of extra bolts
+  efHotHeal: number = 0;//healing from hots
+  efHotOverheal: number = 0;//overhealing from hots
 
-  totalBolts = 0;
+  boltCount: number = 0;
 
-  hotMap = new Map();//tracking on who has what and how it is
+  totalBolts: number = 0;
 
-  masteryTickTock = false;
-  masteryHit = 0;
-  masteryHealing = 0;
-  masteryOverhealing = 0;
-  masteryAbsorbed = 0;
+  hotMap: Map<number, {fullCount: boolean; applicationTime: number;}> = new Map<number, {fullCount: boolean; applicationTime: number;}>();//tracking on who has what and how it is
```

Comment:
When you're defining these types its fine to use the implicit typing, `varName: number = 0;` is not particularly useful. While normally its fine either way, you can see here where allowing it to be inferred automatically will be much less ugly. In the end this is your specs codebase so stylistic stuff is non-blocking